### PR TITLE
Add testing module

### PR DESCRIPTION
**Pull request by tonymeng**
The goal of this PR is to add a new module `@firebase/testing` that is a relatively stand alone module that can be included to test js apps locally.

### CODE REVIEW

File: packages/testing/src/api/index.ts [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+export function initializeAdminApp(options: any): admin.app.App {
+  if (!('databaseName' in options)) {
+    throw new Error('databaseName not specified');
+  }
+  return admin.initializeApp(
+    {
+      credential: new FakeCredentials(),
+      databaseURL: DBURL + '?ns=' + options.databaseName
+    },
+    'app-' + (new Date().getTime() + Math.random())
```
Comment (ryanpbrewster) [inline packages/testing/src/api/index.ts]: I think this `Math.random()` is actually increasing the chance of a name collision. Can we just use the timestamp or an autoincrement id?

File: packages/testing/src/api/index.ts [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    throw new Error('databaseName not specified');
+  }
+  // if options.auth is not present, we will construct an app with auth == null
+  return admin.initializeApp(
+    {
+      credential: new FakeCredentials(),
+      databaseURL: DBURL + '?ns=' + options.databaseName,
+      databaseAuthVariableOverride: options.auth || null
+    },
+    'app-' + (new Date().getTime() + Math.random())
```
Comment (ryanpbrewster) [inline packages/testing/src/api/index.ts]: maybe call this `admin-app-{}`?

File: packages/testing/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "@firebase/testing",
+  "version": "0.1.0",
```
Comment (jshcrowthe) [inline packages/testing/package.json]: Lets make sure we have the `private` value set to `true` for this module. I don't think we want to publish this quite yet?

File: packages/testing/package.json [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  "description": "",
+  "author": "Firebase <firebase-support@google.com> (https://firebase.google.com/)",
+  "main": "dist/cjs/index.js",
+  "browser": "dist/cjs/index.js",
+  "module": "dist/esm/index.js",
+  "scripts": {
+    "dev": "gulp dev",
+    "test": "run-p test:node",
```
Comment (jshcrowthe) [inline packages/testing/package.json]: If you don't have any browser tests you can omit the `npm-run-all` dependency, and replace this `test` script with the one in `test:node`

File: packages/testing/package.json (hunk 3 above)
Comment (tonymeng) [reply packages/testing/package.json]: done

File: packages/testing/package.json (hunk 4 above)
Comment (tonymeng) [reply packages/testing/package.json]: done

**Comment by jshcrowthe** [comment]
Currently, there is no line in the `.github/CODEOWNERS` file for this package, as such you'll always need someone above that scope to approve PRs.

Want to add the team working on this as a line there? Something like:

```
packages/testing  @tonymeng @ryanpbrewster # Others if there are more
```

**Pull request merged**